### PR TITLE
Add Redis client and george task queue

### PR DESCRIPTION
**Pull request by Bruception**
### Description

- Adds a Redis client
- Adds a bull queue for george tasks
- Adds a configuration to control whether or not we add tasks into bull queue

### CODE REVIEW

File: backend/constants/base-configuration.ts [hunk 1]
```diff
@@ -26,6 +26,9 @@ const BASE_CONFIGURATION: MonkeyTypes.Configuration = {
   enableSavingResults: {
     enabled: false,
   },
+  useRedisForBotTasks: {
```
Comment (Bruception) [inline backend/constants/base-configuration.ts:29]: To be removed once we have migrated

File: backend/api/controllers/result.ts [hunk 2]
```diff
@@ -246,12 +251,20 @@ class ResultController {
     if (result.mode === "time" && String(result.mode2) === "60") {
       UserDAO.incrementBananas(uid, result.wpm);
       if (isPb && user.discordId) {
-        BotDAO.updateDiscordRole(user.discordId, result.wpm);
+        if (useRedisForBotTasks) {
```
Comment (Bruception) [inline backend/api/controllers/result.ts:254]: Wondering if we should dual write here. Just to be able to test that redis is receiving the tasks.

File: backend/tasks/george.ts [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+  };
+}
+
+class George {
+  jobQueue: Queue;
+
+  initJobQueue(redisConnection: IORedis.Redis | undefined): void {
+    this.jobQueue = new Queue("george-tasks", {
+      connection: redisConnection,
+      defaultJobOptions: {
```
Comment (Bruception) [inline backend/tasks/george.ts:22]: Are these defaults fine?
Retry attempt delays will be: 4 seconds, 8 seconds, 16 seconds.

File: backend/api/controllers/result.ts (hunk 2 above)
Comment (Miodec) [reply backend/api/controllers/result.ts:254]: Dual write would cause same commands to be executed twice when the configuration is switched

**Review by Miodec — APPROVED** [review]

**Pull request merged**